Modules/TPC/src/CalPadClusterReductor.cxx

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+//
+// file  CalPadClusterReductor.cxx
+// author Marcel Lesch
+//
+
+#include "TPC/CalPadClusterReductor.h"
+#include "TPC/ClustersData.h"
+#include "QualityControl/QcInfoLogger.h"
+#include <TMath.h>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+
+namespace o2::quality_control_modules::tpc
+{
+
+void* CalPadClusterReductor::getBranchAddress()
+{
+  return &mCalPad;
+} // void* CalPadClusterReductor::getBranchAddress()
+
+const char* CalPadClusterReductor::getBranchLeafList()
+{
+  return "NClusters[5][72]/F:QMax[5][72]:QTot[5][72]:SigmaTime[5][72]:SigmaPad[5][72]:TimeBin[5][72]";
+} // const char* CalPadClusterReductor::getBranchLeafList()
+
+void CalPadClusterReductor::update(TObject* obj)
+{
+  if (obj) {
+    auto qcClusters = dynamic_cast<ClustersData*>(obj);
+
+    if (qcClusters) {
+      auto& clusters = qcClusters->getClusters();
+
+      for (int iType = 0; iType < 6; iType++) {
+
+        auto& calDet = GetCalPad(clusters, iType);
+
+        for (size_t iROC = 0; iROC < calDet.getData().size(); ++iROC) {
+          auto& calArray = calDet.getCalArray(iROC);
+          auto& data = calArray.getData();
+
+          // Remove pads which are empty from any calculations
+          data.erase(std::remove_if(data.begin(), data.end(), [](const auto& value) { return (std::isnan(value) || value <= 0); }), data.end());
+
+          Float_t(*Quantity)[72] = getArrayPointer(iType);
+          Quantity[0][iROC] = static_cast<float>(data.size());
+          Quantity[1][iROC] = TMath::Mean(data.begin(), data.end());
+          Quantity[2][iROC] = TMath::StdDev(data.begin(), data.end());
+          Quantity[3][iROC] = TMath::Median(data.size(), data.data());
+          Quantity[4][iROC] = TMath::RMS(data.size(), data.data());
```

Comment:
I think `TMath::RMS` and TMath::Stddev` are identical https://root.cern.ch/doc/master/TMath_8h_source.html#l00525, so no need to have both. Since `StdDev` is the correct term, I would only keep that.

Reply:
Done